### PR TITLE
Better Node Management, Orders Api and Refactoring

### PR DESCRIPTION
**Pull request by sschiessl-bcp**
Well, this escalated quickly..

I tried solving #1521 and add a feature for #1672 and I ended up refactoring SettingsStore and AccessSettings. Which took a massive amount of time, but was not agreed upon so its mostly my own fault. 

Anyways I need to make a cut now and would implement actual filtering options for the user in a separate feature branch in which I also complete updating the apiConfig.

This happened:
- refactoring of SettingsStore, also
    - remove settings default bug that defaults are not overwritten (settings_v4 now) and another loading bug
    - activeNode doesnt need to be in local storage
- Faucet is not editable anymore
- User can now Ping all nodes when his node is out of sync
- refactoring of AccessSettings, also
    - moved all logic into ApiNode and seperated where necessary
- more sophisticated node object with default, hidden etc.
- Introduce new component: ChoiceModal that allows to present the user different choices how to proceed

My actual hours were roughly spent on (roughly):
- Refactoring SettingsStore and AccessSettings 14
- OrdersApi analysis and finding of invalid nodes 4
- Fixing the default settings bug 5
- Better node management 4

How do I translate that into points? I do not expect payment on my refactoring efforts as this was not agreed.

### CODE REVIEW

File: app/stores/SettingsStore.js [hunk 1]
```diff
@@ -257,7 +257,7 @@ class SettingsStore {
         //       set if not present on loading
         let support_v3_until = new Date("2018-10-20T00:00:00Z");
 
-        if (!ss.get("settings_v4") && new Date() < support_v3_until) {
+        if (!ss.has("settings_v4") && new Date() < support_v3_until) {
```
Comment (sschiessl-bcp) [inline app/stores/SettingsStore.js]: Interesting, did not know that method. Will try that out to see the difference, thanks

Comment (svk31) [reply app/stores/SettingsStore.js]: By default get returns an object, that's why your previous check always evaluated to false. !{} is always false, regardless of the number of keys inside.

Comment (sschiessl-bcp) [reply app/stores/SettingsStore.js]: Ah cool. I thought I tried and saw that there is undefined ... Yours makes more sense and is the precise question this line of code wants to ask. Thanks for clarifying.

**Comment by sschiessl-bcp** [comment]
@wmbutler @startailcoon 

Please rate the hours in here. I repeat from the pull request:

- Refactoring SettingsStore and AccessSettings 14: Not based on any issue, see my initial text above
- OrdersApi analysis and finding of invalid nodes 4: https://github.com/bitshares/bitshares-ui/issues/1672
- Fixing the default settings bug 5: https://github.com/bitshares/bitshares-ui/issues/1462
- Better node management 4: https://github.com/bitshares/bitshares-ui/issues/1521

**Comment by startailcoon** [comment]
I have updated the hours for issues #1672 and #1462, as they took longer than estimated, nothing strange about that. They are both large issues.

As I see it, the refactoring is a sub-task of both these issues was required for resolving them. A good practice would probably be to file a new issue task with the details, add the hours and target it to this PR as a closed issue. 

Great work @sschiessl-bcp 

cc @wmbutler @svk31

**Comment by wmbutler** [comment]
Looks good to me as well. Great work.

**Pull request merged**